### PR TITLE
Add nanoflow support on click event

### PR DESCRIPTION
**Pull request by jseremba**

### CODE REVIEW

File: src/BarcodeScannerForPhoneGap/widget/BarcodeScannerForPhoneGap.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ define([
         buttonLabel: "",
         buttonClass: "",
         onchangemf: "",
+        onchangeNanoFlow: "",
```
Comment (akileng56) [inline src/BarcodeScannerForPhoneGap/widget/BarcodeScannerForPhoneGap.js]: Choose one type of casing (camel case) => onChangeNanoflow

File: src/BarcodeScannerForPhoneGap/BarcodeScannerForPhoneGap.xml [hunk 2]
```diff
@@ -31,5 +31,11 @@
             <description></description>
             <returnType type="Void"/>
         </property>
+        <property key="onchangeNanoFlow" type="nanoflow" required="false">
```
Comment (akileng56) [inline src/BarcodeScannerForPhoneGap/BarcodeScannerForPhoneGap.xml]: Change to => onChangeNanoflow

File: src/BarcodeScannerForPhoneGap/widget/BarcodeScannerForPhoneGap.js [hunk 3]
```diff
@@ -55,6 +56,7 @@ define([
             if (!output.cancelled && output.text && output.text.length > 0) {
                 this._obj.set(this.attributeName, output.text);
                 this._executeMicroflow();
+                this._executeNanoflow();
```
Comment (akileng56) [inline src/BarcodeScannerForPhoneGap/widget/BarcodeScannerForPhoneGap.js]: Rename the _executeMicroflow() to be generic maybe executeAction and handle both events from in there

File: package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "BarcodeScannerForPhoneGap",
+  "version": "3.0.1",
+  "description": "",
```
Comment (akileng56) [inline package.json]: Remove what you are not using or fill them in

File: package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "BarcodeScannerForPhoneGap",
+  "version": "3.0.1",
```
Comment (akileng56) [inline package.json]: Check the current version of the Appstore and update accordingly...It's a new feature so, i guess it should be 3.1 now

File: src/BarcodeScannerForPhoneGap/BarcodeScannerForPhoneGap.xml [hunk 6]
```diff
@@ -31,5 +31,11 @@
...
+            <caption>On change nanoflow</caption>
+            <category>Events</category>
+            <description></description>
```
Comment (akileng56) [inline src/BarcodeScannerForPhoneGap/BarcodeScannerForPhoneGap.xml]: If there is no description then use `<description/>` instead

**Review by akileng56 — CHANGES_REQUESTED** [review]

File: src/BarcodeScannerForPhoneGap/widget/BarcodeScannerForPhoneGap.js (hunk 1 above)
Comment (jseremba) [reply src/BarcodeScannerForPhoneGap/widget/BarcodeScannerForPhoneGap.js]: Done

File: src/BarcodeScannerForPhoneGap/widget/BarcodeScannerForPhoneGap.js (hunk 3 above)
Comment (jseremba) [reply src/BarcodeScannerForPhoneGap/widget/BarcodeScannerForPhoneGap.js]: Done

File: package.json (hunk 4 above)
Comment (jseremba) [reply package.json]: Done

File: src/BarcodeScannerForPhoneGap/BarcodeScannerForPhoneGap.xml (hunk 6 above)
Comment (jseremba) [reply src/BarcodeScannerForPhoneGap/BarcodeScannerForPhoneGap.xml]: Done

File: package.json (hunk 5 above)
Comment (jseremba) [reply package.json]: Version Bumped up

File: src/BarcodeScannerForPhoneGap/BarcodeScannerForPhoneGap.xml (hunk 2 above)
Comment (jseremba) [reply src/BarcodeScannerForPhoneGap/BarcodeScannerForPhoneGap.xml]: Fixed

**Pull request merged**